thunderbird/src/main/java/zeta/android/thunderbird/models/transformers/PdpComponentizedModelTransformer.java

```diff
@@ -134,19 +152,229 @@ private PdpV3ServiceabilityCard transformServiceabilityCard(PdpComponentizationC
                 .build();
     }
 
+    //region PdpV3SocialCard transformer methods
+
     private PdpV3SocialCard transformSocialCard(PdpComponentizationCardsResponse socialCard) {
+
+        @PdpV3ComponentType
+        LinkedHashMap<String, Integer> componentPositionIndex = new LinkedHashMap<>(0);
+        PdpV3LikersLazyComponent pdpV3LikersLazyComponent = null;
+        PdpV3CompleteLookComponent pdpV3CompleteLookComponent = null;
+
+        if(CollectionUtils.hasElements(socialCard.componentsList)){
+            final int componentListSize = socialCard.componentsList.size();
+            for(int indexPos = 0; indexPos < componentListSize; indexPos++){
+                PdpComponentizationComponentsResponse pdpComponentizationComponentsResponse
+                        = socialCard.componentsList.get(indexPos);
+                @PdpV3ComponentType
+                final String componentType = pdpComponentizationComponentsResponse.type;
+                switch (componentType){
+                    case PdpV3ComponentType.LIKERS_LAZY:
+                        transformPdpV3LikersLazyComponent(pdpComponentizationComponentsResponse.props.actionType,
+                                pdpComponentizationComponentsResponse.props.action,
+                                pdpComponentizationComponentsResponse.args.title);
+                        componentPositionIndex.put(PdpV3ComponentTypeUtil.from(componentType), indexPos);
+                        break;
+                    case PdpV3ComponentType.COMPLETE_LOOK:
+                        transformPdpV3CompleteLookComponent(pdpComponentizationComponentsResponse.props.styleNote,
+                                pdpComponentizationComponentsResponse.args.title);
+                        componentPositionIndex.put(PdpV3ComponentTypeUtil.from(componentType), indexPos);
```

Comment:
We can pull this out of switch case right `componentPositionIndex.put(PdpV3ComponentTypeUtil.from(componentType), indexPos);` because its common code.